File: JsonResponsePost/app/src/main/java/com/example/jsonresponsepost/MainActivity.java
package com.example.jsonresponsepost;

import android.os.Bundle;
import android.util.Log;
import android.widget.ArrayAdapter;
import android.widget.ListView;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

import com.android.volley.Request;
import com.android.volley.RequestQueue;
import com.android.volley.Response;
import com.android.volley.VolleyError;
import com.android.volley.toolbox.JsonArrayRequest;
import com.android.volley.toolbox.JsonObjectRequest;
import com.android.volley.toolbox.Volley;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ListView listView = findViewById(R.id.listView);

        String url = "https://postman-rest-api-learner.glitch.me//info";

        RequestQueue requestQueue = Volley.newRequestQueue(this);
        ArrayList<String> arrNames = new ArrayList<>();

        JsonObjectRequest jsonObjectRequest = new JsonObjectRequest(
                Request.Method.POST,
                url,
                null,
                new Response.Listener<JSONObject>() {
                    @Override
                    public void onResponse(JSONObject response) {
                        Log.d("MainActivity", "Response: " + response.toString());
                        try {
                            // Loop through each item in the array
                            JSONObject user = response.getJSONObject("message");
                            String userName = user.getString("message");
                            arrNames.add(userName);

                            ArrayAdapter<String> adapter = new ArrayAdapter<String>(MainActivity.this, android.R.layout.simple_list_item_1, arrNames);
                            listView.setAdapter(adapter);
                            // Example: Logging user id and name
                            Log.d("MainActivity",  "Message: " + userName);
                        } catch (JSONException e) {
                            // JSON parsing error
                            Log.e("MainActivity", "Error parsing JSON response", e);
                        }
                    }
                },
                new Response.ErrorListener() {
                    @Override
                    public void onErrorResponse(VolleyError error) {
                        // Handle Volley error
                        Log.e("MainActivity", "Volley error", error);
                    }
                }
        );

        // Add the request to the RequestQueue
        requestQueue.add(jsonObjectRequest);



    }
}

File: RecyclerViewKotlin/app/src/main/java/com/example/recyclerviewkotlin/MainActivity.kt
package com.example.recyclerviewkotlin

import android.os.Bundle
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import androidx.recyclerview.widget.GridLayoutManager
import com.example.recyclerviewkotlin.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    private lateinit var binding: ActivityMainBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        val arrContact = ArrayList<ContactModel>()
        arrContact.add(ContactModel(R.drawable.baseline_person_24,"A","9876543210"))
        arrContact.add(ContactModel(R.drawable.baseline_person_24,"B","9876543210"))
        arrContact.add(ContactModel(R.drawable.baseline_person_24,"C","9876543210"))
        arrContact.add(ContactModel(R.drawable.baseline_person_24,"D","9876543210"))
        arrContact.add(ContactModel(R.drawable.baseline_person_24,"E","9876543210"))
        arrContact.add(ContactModel(R.drawable.baseline_person_24,"F","9876543210"))
        arrContact.add(ContactModel(R.drawable.baseline_person_24,"G","9876543210"))
        arrContact.add(ContactModel(R.drawable.baseline_person_24,"H","9876543210"))

        binding.recyclerView.layoutManager = GridLayoutManager(this,4)
        binding.recyclerView.adapter = RecyclerContactAdapter(this,arrContact)
    }
}